src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/excel/MultipleCloseEventValidationService.java

```diff
@@ -31,24 +37,43 @@ public MultipleCloseEventValidationService(SingleCasesReadingService singleCases
 
     public List<String> validateCasesBeforeCloseEvent(String userToken, MultipleDetails multipleDetails) {
         List<String> errors = new ArrayList<>();
-        var multipleData = multipleDetails.getCaseData();
-
-        List<String> ethosCaseRefCollection = multipleHelperService.getEthosCaseRefCollection(userToken, multipleData,
-                errors);
+        var ethosCaseRefCollection = multipleHelperService.getEthosCaseRefCollection(userToken,
+                multipleDetails.getCaseData(), errors);
 
         if (ethosCaseRefCollection.isEmpty()) {
             return errors;
-        }
+        } else {
+            var submitEvents = singleCasesReadingService.retrieveSingleCases(userToken,
+                    multipleDetails.getCaseTypeId(), ethosCaseRefCollection,
+                    multipleDetails.getCaseData().getMultipleSource());
 
-        List<SubmitEvent> submitEvents = singleCasesReadingService.retrieveSingleCases(userToken,
-                multipleDetails.getCaseTypeId(), ethosCaseRefCollection, multipleData.getMultipleSource());
-
-        for (SubmitEvent event : submitEvents) {
-            var caseData = event.getCaseData();
-            eventValidationService.validateCaseBeforeCloseEvent(caseData, event.getState().equals(REJECTED_STATE),
-                    true, errors);
+            for (var submitEvent : submitEvents) {
+                validateCase(submitEvent, errors);
+                if (!errors.isEmpty()) {
+                    break;
+                }
+            }
         }
 
         return errors;
     }
+
+    private void validateCase(SubmitEvent submitEvent, List<String> errors) {
+        for (var conditionToValidate : validationConditions) {
+
+            if (JURISDICTION.equals(conditionToValidate)) {
+                eventValidationService.validateJurisdictionOutcome(submitEvent.getCaseData(),
+                        submitEvent.getState().equals(REJECTED_STATE), true, errors);
+            } else if (HEARING_STATUS.equals(conditionToValidate)) {
+                eventValidationService.validateHearingStatusForCaseCloseEvent(submitEvent.getCaseData(), errors);
+            } else if (JUDGE_ALLOCATION.equals(conditionToValidate)) {
+                eventValidationService.validateHearingJudgeAllocationForCaseCloseEvent(submitEvent.getCaseData(),
+                        errors);
+            }
+
+            if (!errors.isEmpty()) {
+                break;
+            }
```

Comment:
This seems to be needlessly convoluted? The way you've coded the validateCaseForCaseCloseEvent in the single case validations is fine?

Reply:
point taken